medium/profiles/services.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any, BinaryIO, Dict, Optional
+
+from django.core.exceptions import ValidationError as DjangoValidationError
+from rest_framework.exceptions import ValidationError as DRFValidationError
+
+from medium.common.services import model_update
+from medium.profiles.models import Profile
+from medium.users.models import User
+
+
+class ProfileService:
+    def __init__(self, user_id: int, profile_pic: BinaryIO, header_pic: BinaryIO):
+        self.user_id = user_id
+        self.profile_pic = profile_pic
+        self.header_pic = header_pic
+
+    def create_profile(
+        self,
+        about_text: str,
+        **optional_fields: Optional[Dict[str, Any]],
```

Comment:
I guess hardcoding the fields is less error-prone,
if a certain field is an optional type hint it then add a default value for it

Reply:
I think the current signature is more succinct and is consistent with the user services, However I agree that hardcoding the arguments will make it more convenient for the caller which may reduce bugs ... 

I'll be changing that